.github/workflows/reusable-san.yml

```
@@ -65,30 +81,30 @@ jobs:
       run: >-
         ./configure
         --config-cache
-        --with-thread-sanitizer
+        ${{ inputs.sanitizer == 'TSan' && '--with-thread-sanitizer' || '--with-undefined-behavior-sanitizer' }}
         --with-pydebug
         ${{ fromJSON(inputs.free-threading) && '--disable-gil' || '' }}
     - name: Build CPython
       run: make -j4
     - name: Display build info
       run: make pythoninfo
     - name: Tests
-      run: ./python -m test --tsan -j4
+      run: ./python -m test ${{ inputs.sanitizer == 'TSan' && '--tsan' || '' }} -j4
     - name: Parallel tests
       if: fromJSON(inputs.free-threading)
```

This should probably be guarded by tsan too:
```suggestion
      if: >-
        inputs.sanitizer == 'TSan'
        && fromJSON(inputs.free-threading)
```